solana/solana-rpc/src/rpc.ts

```diff
@@ -132,6 +141,23 @@ export class Rpc implements RpcApi {
 
         return pack.flat()
     }
+
+    validateGetBlockResult(result: unknown, req: RpcRequest) {
+        let validator = getResultValidator(nullable(GetBlock))
+        let block = validator(result)
+        if (this.txThreshold && block != null && block.transactions != null) {
+            let transactions = block.transactions.filter(tx => !isVoteTransaction(tx))
```

Comment:
my bad i didn't understand my measurements at first. it doesn't take ~30ms to filter it's just ~0.030ms in average